pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -1812,7 +1813,7 @@ private boolean updateExistedSegment(String tableNameWithType, SegmentZKMetadata
    * Returns a map from server instance to list of segments it serves for the given table.
    */
   public Map<String, List<String>> getServerToSegmentsMap(String tableNameWithType) {
-    Map<String, List<String>> serverToSegmentsMap = new HashMap<>();
+    Map<String, List<String>> serverToSegmentsMap = new TreeMap<>();
```

Comment:
why change this to TreeMap? This method is also used by table size reader, and when servers have tens of thousands of segments, it will add significant overhead.
If necessary, the API to get the server map can sort it (for disaplay purpose)

Reply:
ah, the sort is on the servers, so ignore this